nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateJson.java

```diff
@@ -79,65 +92,69 @@
-            this.versionFlag = versionFlag;
-        }
 
-        @Override
-        public String getValue() {
-            return name();
-        }
+    public static final String SCHEMA_NAME_PROPERTY_NAME = "Schema Name";
+    public static final String SCHEMA_CONTENT_PROPERTY_NAME = "JSON Schema";
+    public static final String ERROR_ATTRIBUTE_KEY = "json.validation.errors";
+    public static final AllowableValue SCHEMA_NAME_PROPERTY = new AllowableValue("schema-name", "Use '" + SCHEMA_NAME_PROPERTY_NAME + "' Property",
+            "The name of the Schema to use is specified by the '" + SCHEMA_NAME_PROPERTY_NAME +
+                    "' Property. The value of this property is used to lookup the Schema in the configured JSON Schema Registry service.");
+    public static final AllowableValue SCHEMA_CONTENT_PROPERTY = new AllowableValue("schema-content-property", "Use '" + SCHEMA_CONTENT_PROPERTY_NAME + "' Property",
+            "A URL or file path to the JSON schema or the actual JSON schema is specified by the '" + SCHEMA_CONTENT_PROPERTY_NAME + "' Property. " +
+                    "No matter how the JSON schema is specified, it must be a valid JSON schema");
 
-        @Override
-        public String getDisplayName() {
-            return displayName;
-        }
+    private static final List<AllowableValue> STRATEGY_LIST = Arrays.asList(SCHEMA_NAME_PROPERTY, SCHEMA_CONTENT_PROPERTY);
 
-        @Override
-        public String getDescription() {
-            return description;
-        }
+    public static final PropertyDescriptor SCHEMA_ACCESS_STRATEGY = new PropertyDescriptor.Builder()
+            .name("schema-access-strategy")
+            .displayName("Schema Access Strategy")
+            .description("Specifies how to obtain the schema that is to be used for interpreting the data.")
+            .allowableValues(STRATEGY_LIST.toArray(new AllowableValue[0]))
+            .defaultValue(SCHEMA_CONTENT_PROPERTY.getValue())
+            .required(true)
+            .build();
 
-        public VersionFlag getVersionFlag() {
-            return versionFlag;
-        }
-    }
+    public static final PropertyDescriptor SCHEMA_NAME = new PropertyDescriptor.Builder()
+            .name("schema-name")
+            .displayName(SCHEMA_NAME_PROPERTY_NAME)
+            .description("Specifies the name of the schema to lookup in the Schema Registry property")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .defaultValue("${schema.name}")
+            .dependsOn(SCHEMA_ACCESS_STRATEGY, SCHEMA_NAME_PROPERTY)
+            .required(false)
+            .build();
 
-    public static final String ERROR_ATTRIBUTE_KEY = "json.validation.errors";
+    public static final PropertyDescriptor SCHEMA_REGISTRY = new PropertyDescriptor.Builder()
+            .name("schema-registry")
+            .displayName("Schema Registry")
```

Comment:
For differentiation, it is probably best to include `JSON` in the property name:
```suggestion
            .name("JSON Schema Registry")
            .displayName("JSON Schema Registry")
```